ql/src/java/org/apache/hadoop/hive/ql/exec/tez/DagUtils.java

```diff
@@ -1221,9 +1231,20 @@ private Map<String, LocalResource> addTempResources(Configuration conf, String h
         LOG.info("Skipping vertex resource " + file + " that already exists in the session");
         continue;
       }
-      Path hdfsFilePath = new Path(hdfsDirPathStr, getResourceBaseName(new Path(file)));
-      LocalResource localResource = localizeResource(new Path(file),
-          hdfsFilePath, type, conf);
+      Path filePath = new Path(file);
+      Path hdfsFilePath = new Path(hdfsDirPathStr, getResourceBaseName(filePath));
+      String sha = getSha(filePath, conf);
+      LocalResource localResource = null;
+      boolean alreadyLocalized = localizedJarSha.containsKey(sha);
+      LOG.debug("Checking SHA {} for filePath: {} (temp), already localized: {}", sha, filePath, alreadyLocalized);
+      if (alreadyLocalized) {
+        LOG.info("Skip creating already localized jar (temp) by sha {}: {}", sha, localResource);
+        localResource = localizedJarSha.get(sha);
```

Comment:
oops, I need to assign localResource before logging it :) will fix in next patch